### PR TITLE
Adding latex gas mask and ball gloves. ERP stuff.

### PR DESCRIPTION
**Pull request by Lamella-0587**
## About The Pull Request

This PR adds two new cloth items, that in the future will be used in the new kink vending machine. For now, it's unobtainable but works fine.

## Why It's Good For The Game

This clothing is basically ERP things, that players can use in role interaction and ERP stuff. This part is restraints, that can be used for an interesting raping experience.

Latex gas mask will work as a muzzle, that can't be unequipped without help, but that works as a gasmask and muzzle at the same time. 

Ball mittens are basically a new type of restraints. They can be put in the gloves slot and works as a straightjacket, but can be removed really fast.

Also, there is a reinforced version of them. To make them reinforced players should use handcuffs on the normal version. The reinforced version works as a little bit weaker version of handcuffs, but it's harder to escape than from the basic version. It is used to prevent the character from escaping, while they playing the role of victim, but allowing them to try it without accidental restraints removal.

It's only a little piece of stuff I sprited but I will add more in future PR's

## Changelog
Added latex gas mask and sprites for it
Added ball mittens as a new type of restraints and sprites for it
Added high heels, which slows wearer.

### CODE REVIEW

**Comment by Useroth** [comment]
We'll modularize it properly in near future.

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 1]
```diff
@@ -187,3 +187,36 @@
...
+        user.put_in_hands(W)
+        to_chat(user, "<span class='notice'>You reinforce belts with using [src] on [I].</span>")
+        qdel(I)
+        qdel(src)
+        return
+    . = ..()
+
+/obj/item/clothing/gloves/ball_mittens_rape
+	name = "Reinforced ball mittens"//For raping and BDSM
+	desc = "Do not put these on. Its REALLY hard to put them off... But they looks so comfortable"
```
Comment (KasparoVy) [inline code/modules/clothing/gloves/miscellaneous.dm:215]: But they look*

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -271,6 +271,9 @@
 /mob/living/carbon/is_muzzled()
```
Comment (KasparoVy) [inline code/modules/mob/living/carbon/carbon.dm:271]: Does both this check & the new check below fully function? Hopefully the dual definitions play nice with eachother.

File: code/modules/clothing/gloves/miscellaneous.dm (hunk 1 above)
Comment (Useroth) [reply code/modules/clothing/gloves/miscellaneous.dm:215]: take them off*
as well

Comment (KathrinBailey) [reply code/modules/clothing/gloves/miscellaneous.dm:215]: Line 213, I dunno, something about that type path sounds like a bad idea to be called that. Andm aybe something in 214, in the comment.

Comment (Lamella-0587) [reply code/modules/clothing/gloves/miscellaneous.dm:215]: Sorry for the grammatic, English is not my native language. Feel free to change the description if you need

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 3]
```diff
@@ -187,3 +187,36 @@
 	name = "CentCom intern envirogloves"
 	icon_state = "internplasma"
 	inhand_icon_state = "internplasma"
+
+/obj/item/clothing/gloves/ball_mittens
+	name = "Ball mittens"//For erotic games
```
Comment (atakiya) [inline code/modules/clothing/gloves/miscellaneous.dm:192]: Unneeded comment.

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 4]
```diff
@@ -187,3 +187,36 @@
...
+/obj/item/clothing/gloves/ball_mittens_rape
+	name = "Reinforced ball mittens"//For raping and BDSM
+	desc = "Do not put these on. Its REALLY hard to put them off... But they looks so comfortable"
+	icon_state = "ballmittens"
+	inhand_icon_state = "ballmittens"
+	clothing_flags = DANGEROUS_OBJECT
+	equip_delay_other = 80
+	equip_delay_self = 80
+	strip_delay = 80
+	breakouttime = 20
```
Comment (atakiya) [inline code/modules/clothing/gloves/miscellaneous.dm:222]:
```suggestion
	breakouttime = 20

```
Missing a trailing newline.

File: code/modules/clothing/masks/gasmask.dm [hunk 5]
```diff
@@ -214,6 +214,28 @@
 	desc = "Gnash gnash."
 	icon_state = "carp_mask"
 
+/obj/item/clothing/mask/gas/bdsm_mask
+	name = "Latex gasmask"
+	desc = "Toned gas mask. Completely muffles the wearer, making even breathing really hard with this on." //That's some fetish stuff
```
Comment (atakiya) [inline code/modules/clothing/masks/gasmask.dm:219]: Another unneeded comment

**Comment by Owai-Seek** [comment]
Doesn't git usually get angy about this kind of stuff?

**Comment by Gandalf2k15** [comment]
> Doesn't git usually get angy about this kind of stuff?

Yes. @Azarak please address this

**Comment by hugmeorelse** [comment]
Do not ping @Azarak for this. Let @Useroth handle this.

**Comment by hugmeorelse** [comment]
Is there any possible way we can have it be like PAIs where one can pair up with a friend or even a stranger voluntarily? It sounds like it could be a fun gimmick that way. They can either ask ghosts to be their split personality, select a friend, or maybe choose from a list of potential candidates?

**Comment by Gandalf2k15** [comment]
Howdiddlydoo, I'm seeing a lot of modularisation failures here, and... possible issues down the line. Useroth will be handling these things in future, so for now, I'm going to close this pending his acting. We can't have this in our game, I'll leave it at that for now. Please enquire on discord about this issue.

**Pull request closed without merging**